starttls_policy/tests/util.py

```diff
@@ -0,0 +1,18 @@
+"""Various utils used by tests"""
+
+def parameterized(func):
+    """Partially applies params to function so it can be added
+    as test method to unittest.TestCase class"""
+    def applied_params(*args, **kwargs):
+        def class_method(self):
+            return func(self, *args, **kwargs)
+        return class_method
+    return applied_params
+
+def parameterize_over(cls, test, data):
+    """Links parameter-ready tests with its dataset"""
+    for entry in data:
+        tname, args = entry[0], entry[1:]
+        test_name = "test_" + tname
+        wrapped_test = test(*args)
+        setattr(cls, test_name, wrapped_test)
```

Comment:
Rather than have an API that requires two calls (one to `partial` the args except for the class instance, and one to bind the functions to the class) I would prefer the API to perform a single call.

I'm not sure how you could do this with a single decorator, but maybe we could do it with just `parametrize_over`. That is, would it be possible to define `config_test` as a member function of `TestPostfixGenerator` and have`parametrize_over` apply the partial args?

Something like the below might work, as long as `test` is bound to an instance of `cls`.
```suggestion
        def _partial(func, *args, **kwargs):
          def wrapped(self):
            return func(self, *args, **kwargs)
          return wrapped
        setattr(cls, test_name, _partial(test, *args))
```

Reply:
fixed